compiler/rustc_codegen_gcc/src/intrinsic/mod.rs

```diff
@@ -309,6 +309,8 @@ impl<'a, 'gcc, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'a, 'gcc, 'tcx> {
                     return;
                 }
 
+                sym::ptr_mask => self.and(args[0].immediate(), args[1].immediate()),
```

Comment:
@antoyo Is this impl correct?

Reply:
I believe it's gonna fail at run-time, but we can keep it like this since I'm not sure how to do pointer masking in GCC yet.

Reply:
Yes, but it would be against the point, no?

Reply:
It would reduce performance, but at least it works.

Reply:
Oh, so it's not related to pointer provenance?

Reply:
It is. But if the backend doesn't support maintaining provenance, then falling back to ptr-int-casts is always okay.

Reply:
Ok, so I can confirm that the and operation doesn't work on pointers in libgccjit, so please use pointer casts.

Reply:
> But if the backend doesn't support maintaining provenance, then falling back to ptr-int-casts is always okay.

Would it be better to put the fallback in `rustc_codegen_ssa`, then?

Reply:
We really should just support defining defaults for intrinsics, like https://github.com/rust-lang/rust/issues/93145 suggests, but someone needs to implement this 🥲 

> so please use pointer casts

I'm not at all familiar with gcc codegen, will something like this work?
```rust
let usize_type = self.context.new_type::<usize>();
let void_ptr_type = self.context.new_type::<*const ()>();

let ptr = args[0].immediate();
let mask = args[1].immediate();

let addr = self.bitcast(ptr, usize_type);
let masked = self.and(addr, mask);
self.bitcast(masked, void_ptr_type)
```

Reply:
Yes, that should work assuming that usize has the same size as a pointer.